### PR TITLE
docs(semantic-release): implement an automated changelog

### PR DESCRIPTION
**Pull request by daks**
I think that all files are present. Please let me know if something is missing.
Also I don't know if there is some configuration to be done on the repo itself.

### CODE REVIEW

**Comment by myii** [comment]
Couple of things for now (which are related by release version number):

1. This will need a `FORMULA` file added.
1. Also need to decide the version number to "start" from.
    1. According to the spreadsheet, the comparative maturity of this formula is `v0.16.0`.
    1. However, there's already a [`CHANGELOG`](https://github.com/saltstack-formulas/collectd-formula/blob/master/CHANGELOG.rst), which is indicating `v1.1.0` or thereabouts -- there is a big jump in there, though -- and it hasn't been updated for years.

---

@n-rodriguez I hope you don't mind me requesting that you review this PR.  We've already implemented `semantic-release` on a couple of formulas (mainly the `template-formula`) and it is working really well.  However, it does require specific commit message formatting, so your input will be valuable.

**Comment by myii** [comment]
@n-rodriguez Just to be clear, I'm not expecting you to go through the actual code changes as such, although you're welcome to comment on that as well.  The main reason is to state your approval/disapproval regarding implementing `semantic-release` into this formula.

**Comment by n-rodriguez** [comment]
> @n-rodriguez I hope you don't mind me requesting that you review this PR.

no problem :+1: 

I'm not against the idea of semantic versioning and the `semantic-release` package.

About the version to start from, I'm not against a "reset' to 1.0.0 : the current changelog is meaningless, the formula is globally stable although it needs more documentation.

 Some states are not documented and the `pillar.example` may need some more examples. I don't know about the tests, but it would worth to check them before we could release a true 1.0.0 version.

**Comment by n-rodriguez** [comment]
About the release process : how does it work?

**Comment by myii** [comment]
@n-rodriguez If you can join the `#formulas` channel in Slack, I can explain it to you in conversation.  Otherwise, you can get the general gist here: https://github.com/saltstack-formulas/template-formula/blob/master/docs/CONTRIBUTING.rst#commit-message-formatting.

**Comment by daks** [comment]
Hi,

about the `FORMULA` file, isn't it auto-generated by the versioning script? How do we maintain the version number inside?
About the version number to publish, I have no idea what's the best: are actual numbers valid and used? If we don't want to mess with that, create a 2.0 can also be possible. Or a 1.2.

**Comment by myii** [comment]
Just copying this over from our discussion on Slack:

> @daks About the `FORMULA` file, the basic arrangement needs to be provided.  Just borrow the one in `template-formula` and modify accordingly.  Only the version number is updated automatically during the release process.
> As for the version number to start from, then the spreadsheet suggests `v0.1.6`.  However, the out-of-date Changelog suggests `v1.1.0`.  Just put either for now and we can sort it out when finalising this PR.

**Comment by daks** [comment]
`FORMULA` file added

File: FORMULA [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: collectd
+os: Debian, Ubuntu, Raspbian, RedHat, Fedora, CentOS, Suse, openSUSE, Gentoo, Funtoo, Arch, Manjaro, Alpine, FreeBSD, OpenBSD, Solaris, SmartOS, Windows, MacOS
+os_family: Debian, RedHat, Suse, Gentoo, Arch, Alpine, FreeBSD, OpenBSD, Solaris, Windows, MacOS
+version: 1.0rc
+release: 1
+minimum_version: 2016.11
+summary: Template formula
```
Comment (myii) [inline FORMULA]: Not `Template` formula.

File: docs/CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+An example of that:
+
+.. code-block:: git
+
+   ...
+
+   BREAKING CHANGE: With the removal of all of the `.sls` files under
+   `template package`, this formula no longer supports the installation of
+   packages.
+
```
Comment (myii) [inline docs/CONTRIBUTING.rst]: Please remove the whole `Documentation` section below.

Comment (myii) [reply docs/CONTRIBUTING.rst]: As in, all the way to the end of the file.  We're not setting up ReadTheDocs for this yet.

File: docs/_static/css/custom.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (myii) [inline docs/_static/css/custom.css]: Please remove this file.

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
```
Comment (myii) [inline docs/conf.py]: Please remove this file.

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,20 @@
+.. _index:
```
Comment (myii) [inline docs/index.rst]: Please remove this file.

**Comment by myii** [comment]
@daks Thanks for all of your work with this.  Great job!

**Comment by myii** [comment]
@daks Merged, finally!  The new release should be delivered in the next few minutes...

**Comment by daks** [comment]
@myii thanks for your assistance

**Comment by myii** [comment]
@daks You're welcome.  Still a few remaining tasks to complete this properly (put these as issues?):

- [ ] Merge the Changelog files into one (and maybe fill in more of the history?).
- [ ] Use this content as the entry for release `v1.0.0` (the one before `semantic-release`).
- [ ] Upgrade the CI matrix to use the pre-salted images -- the current run takes over 20 minutes (over 45 minutes of total time).

**Pull request merged**